cpp/include/raft/spatial/knn/knn.cuh

```diff
@@ -122,16 +121,17 @@ enum class SelectKAlgo {
  *   the implementation of the algorithm
  */
 template <typename idx_t = int, typename value_t = float>
-inline void select_k(const value_t* in_keys,
-                     const idx_t* in_values,
-                     size_t n_inputs,
-                     size_t input_len,
-                     value_t* out_keys,
-                     idx_t* out_values,
-                     bool select_min,
-                     int k,
-                     cudaStream_t stream,
-                     SelectKAlgo algo = SelectKAlgo::FAISS)
+[[deprecated("Use function `select_k` from `raft/matrix/select_k.cuh")]] inline void select_k(
```

Comment:
We should also add a small note directly to the doxygen for this as well ("Note: This call is deprecated, please use `raft/matrix/select_k.cuh`").